README.md

```diff
@@ -24,15 +23,77 @@ module.exports = ({ env }) => ({
       secretAccessKey: env('AWS_ACCESS_SECRET'),
       region: env('AWS_REGION'),
       params: {
-        Bucket: env('AWS_BUCKET'),
+        bucket: env('AWS_BUCKET'),
       },
       baseUrl: env('CDN_BASE_URL'), // e.g. https://cdn.example.com, this is stored in strapi's database to point to the file
-      prefix: env('BUCKET_PREFIX') // e.g. strapi-assets, note the missing slash at the start
+      prefix: env('BUCKET_PREFIX'), // e.g. strapi-assets, note the missing slash at the start
+    },
+  },
+  // ...
+});
+```
+
+### Strapi v4
+
+If using >4.0.0, please use the below config:
+
+`./config/plugins.js`
+```js
+module.exports = ({ env }) => ({
+  // ...
+  upload: {
+    config: {
+      provider: 'strapi-provider-upload-aws-s3-advanced',
+      providerOptions: {
+        accessKeyId: env('AWS_ACCESS_KEY_ID'),
+        secretAccessKey: env('AWS_ACCESS_SECRET'),
+        region: env('AWS_REGION'),
+        params: {
+          bucket: env('AWS_BUCKET'),
+        },
+        baseUrl: env('CDN_BASE_URL'), // e.g. https://cdn.example.com, this is stored in strapi's database to point to the file
+        prefix: env('BUCKET_PREFIX'), // e.g. strapi-assets, note the missing slash at the start
+      },
     },
   },
   // ...
 });
 ```
+#### Image Previews
+
+To allow the thumbnails to properly populate, add the below config to
+
+`./config/plugins.js`
```

Comment:
```suggestion
`./config/middlewares.js`
```

According to <https://github.com/strapi/strapi/tree/master/packages/providers/upload-aws-s3>, this should be in `./config/middlewares.js`, which also resolves my confussion regarding the return type of the function exported from plugins.js

Reply:
You're absolutely right - this was just an oversight on my part.